### PR TITLE
Dubbo samples mesh k8s: add envoy healthChecks

### PR DESCRIPTION
**Pull request by conghuhu**
feat: add envoy healthChecks to dubbo-samples-mesh-k8s.
fix: edit Service.yml to apply for grpc loadbalance.

### CODE REVIEW

File: dubbo-samples-mesh-k8s/dubbo-samples-mesh-consumer/pom.xml [hunk 1]
```diff
@@ -30,11 +30,11 @@
     <properties>
         <source.level>1.8</source.level>
         <target.level>1.8</target.level>
-        <dubbo.version>3.1.0-SNAPSHOT</dubbo.version>
+        <dubbo.version>3.0.7</dubbo.version>
```
Comment (CrazyHZM) [inline dubbo-samples-mesh-k8s/dubbo-samples-mesh-consumer/pom.xml:33]: Don't change this version.

File: dubbo-samples-mesh-k8s/dubbo-samples-mesh-consumer/pom.xml [hunk 2]
```diff
@@ -63,6 +63,7 @@
             <groupId>org.apache.dubbo</groupId>
             <artifactId>dubbo</artifactId>
             <version>${dubbo.version}</version>
+            <!--            <version>3.1.0-SNAPSHOT</version>-->
```
Comment (CrazyHZM) [inline dubbo-samples-mesh-k8s/dubbo-samples-mesh-consumer/pom.xml:66]: Remove useless code.

File: dubbo-samples-mesh-k8s/dubbo-samples-mesh-consumer/src/main/java/org/apache/dubbo/samples/ConsumerBootstrap.java [hunk 3]
```diff
@@ -19,23 +19,28 @@
...
         GreetingServiceConsumer greetingServiceConsumer = context.getBean(GreetingServiceConsumer.class);
-        greetingServiceConsumer.doSayHello("Kubernetes Api Server");
-        System.out.println("==================== dubbo invoke end ====================");
+        AtomicInteger count = new AtomicInteger(0);
+        while (true) {
+            greetingServiceConsumer.doSayHello("service mesh");
+            System.out.println("==================== dubbo invoke " + count.get() + " end ====================");
+            count.getAndIncrement();
+            Thread.sleep(10000);
+        }
```
Comment (CrazyHZM) [inline dubbo-samples-mesh-k8s/dubbo-samples-mesh-consumer/src/main/java/org/apache/dubbo/samples/ConsumerBootstrap.java:43]: Don't loop endlessly.

File: dubbo-samples-mesh-k8s/dubbo-samples-mesh-consumer/src/main/java/org/apache/dubbo/samples/action/GreetingServiceConsumer.java [hunk 4]
```diff
@@ -33,13 +33,14 @@ public class GreetingServiceConsumer {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GreetingServiceConsumer.class);
 
-    @DubboReference(version = "1.0.0", providedBy = "dubbo-samples-mesh-provider", url = "tri://dubbo-samples-mesh-provider:50052")
+    @DubboReference(version = "1.0.0", providedBy = "dubbo-samples-mesh-provider", url = "tri://dubbo-samples-mesh-provider:50052", lazy = true)
+//    @DubboReference(version = "1.0.0", providedBy = "dubbo-samples-mesh-provider", url = "tri://localhost:50052")
```
Comment (CrazyHZM) [inline dubbo-samples-mesh-k8s/dubbo-samples-mesh-consumer/src/main/java/org/apache/dubbo/samples/action/GreetingServiceConsumer.java:37]: Remove useless code.

File: dubbo-samples-mesh-k8s/dubbo-samples-mesh-provider/pom.xml [hunk 5]
```diff
@@ -11,8 +11,8 @@
     <properties>
         <source.level>1.8</source.level>
         <target.level>1.8</target.level>
-        <dubbo.version>3.1.0-SNAPSHOT</dubbo.version>
-        <grpc.version>1.44.0</grpc.version>
+        <dubbo.version>3.0.7</dubbo.version>
+        <grpc.version>1.31.1</grpc.version>
```
Comment (CrazyHZM) [inline dubbo-samples-mesh-k8s/dubbo-samples-mesh-provider/pom.xml:15]: Don't change this version.

File: dubbo-samples-mesh-k8s/dubbo-samples-mesh-consumer/pom.xml [hunk 6]
```diff
@@ -30,11 +30,11 @@
     <properties>
         <source.level>1.8</source.level>
         <target.level>1.8</target.level>
-        <dubbo.version>3.1.0-SNAPSHOT</dubbo.version>
+        <dubbo.version>3.0.7</dubbo.version>
         <spring.version>4.3.16.RELEASE</spring.version>
         <junit.version>4.12</junit.version>
         <maven-compiler-plugin.version>3.7.0</maven-compiler-plugin.version>
-        <grpc.version>1.44.0</grpc.version>
+        <grpc.version>1.31.1</grpc.version>
```
Comment (CrazyHZM) [inline dubbo-samples-mesh-k8s/dubbo-samples-mesh-consumer/pom.xml:37]: Don't change this version.

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

**Comment by chickenlj** [comment]
Please submit followup patches to fix the above minor issues.

**Pull request merged**